src/connectedk8s/azext_connectedk8s/_params.py

```diff
@@ -32,8 +32,12 @@ def load_arguments(self, _):
         c.argument('proxy_cert', options_list=['--proxy-cert'], arg_group='Proxy', type=file_type, completer=FilesCompleter(), help='Path to the certificate file for proxy')
         c.argument('distribution', options_list=['--distribution'], help='The Kubernetes distribution which will be running on this connected cluster.', arg_type=get_enum_type(Distribution_Enum_Values))
         c.argument('infrastructure', options_list=['--infrastructure'], help='The infrastructure on which the Kubernetes cluster represented by this connected cluster will be running on.', arg_type=get_enum_type(Infrastructure_Enum_Values))
-        c.argument('disable_auto_upgrade', options_list=['--disable-auto-upgrade'], action='store_true', help='Flag to disable auto upgrade of arc agents.')
```

Comment:
let's make this --auto-upgrade true/false instead of --disable-auto-upgrade so that we use the same param name across connect and update. we can have true as default value for --auto-upgrade

Reply:
also let's add this to the arg_group= 'Agent upgrade and container registry"

Reply:
ack

Reply:
Let's not do this for now. This basically means we are deprecating a CLI argument and we need to follow certain [guidelines](https://github.com/Azure/azure-cli/blob/29767d75d850ddc1c24cc85bd46d861b61d77a47/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments) to do this. Worth discussing with CLI team once and ask their guidance before taking this change. @alphaWizard to confirm if there was a reason to keep this as --disable-auto-upgrade

Reply:
I think, this choice was made at that time, because we wanted the default behaviour during the connect to enable auto-update. And override that using --disable-auto-upgrade.

And, toggle later using update command using --auto-upgrade true/false

Reply:
@santosh02iiit , Can you follow the deprecation guidelines for this? And see how it reflects in --help of the connect command for user experience scenarios.